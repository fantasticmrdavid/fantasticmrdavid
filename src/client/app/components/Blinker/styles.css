.Blinker {
  position: relative;
  display: inline-block;
  width: 0.5rem;
  margin-left: 0.15rem;
  animation: blink 1s step-end infinite;
  border-bottom: 2px solid #aaa;
}

@keyframes blink {
  0% { opacity: 1.0; }
  50% { opacity: 0.0; }
  100% { opacity: 1.0; }
}
