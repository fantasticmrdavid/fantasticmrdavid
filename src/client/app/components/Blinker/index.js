import './styles.css';
import React from 'react';

export default () => {
  return <div className="Blinker"></div>
}
