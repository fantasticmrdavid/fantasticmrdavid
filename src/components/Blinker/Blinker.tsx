import React, { memo } from 'react';
import { Cursor } from './blinker.styles';

export const Blinker = memo(() => <Cursor />);
