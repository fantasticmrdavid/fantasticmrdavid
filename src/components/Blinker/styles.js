import styled, { css } from 'styled-components';
import { blink } from 'styles/animations';
import theme from 'styles/theme';

export const Cursor = styled.div`
  position: relative;
  display: inline-block;
  width: 0.5rem;
  background-color: ${theme.colorFont};
  margin-left: 0.15rem;
  animation: ${css`${blink} 1s step-end infinite`};
  border-bottom: 2px solid ${theme.colorFont};
`;
