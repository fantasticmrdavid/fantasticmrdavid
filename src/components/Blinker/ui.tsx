import React, { memo } from 'react';
import * as styles from './styles';

const { Cursor } = styles;

const Blinker = memo(() => <Cursor />);

export default Blinker;
