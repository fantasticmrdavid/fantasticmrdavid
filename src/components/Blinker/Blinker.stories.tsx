import React from "react";
import { Blinker } from "./Blinker";
import { ComponentStory, Meta } from "@storybook/react";

export default {
  title: "Blinker",
  component: Blinker
} as Meta

const Template: ComponentStory<typeof Blinker> = () => <Blinker />;

export const Default = Template.bind({});
