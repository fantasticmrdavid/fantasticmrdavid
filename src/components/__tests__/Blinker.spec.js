import React from 'react';
import renderer from 'react-test-renderer';
import Blinker from 'components/Blinker';

describe('Component -> Blinker', () => {
  it('should render the Blinker without throwing an error', () => {
    expect(renderer.create(<Blinker />)).toMatchSnapshot();
  });
});
