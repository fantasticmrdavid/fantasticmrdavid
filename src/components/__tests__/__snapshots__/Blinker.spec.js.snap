// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component -> Blinker should render the Blinker without throwing an error 1`] = `
<div
  className="styles__Cursor-sc-12n3825-0 Sphwy"
/>
`;
